leader-election-impl/src/main/java/com/palantir/paxos/SqlitePaxosStateLog.java

```diff
@@ -27,91 +28,109 @@
-        boolean writeRound(@Define("table") String table, @Bind("seq") long seq, @Bind("value") byte[] value);
-
-        @SqlQuery("SELECT val FROM <table> WHERE seq = :seq")
+        @SqlUpdate("CREATE TABLE IF NOT EXISTS paxosLog ("
+                + "namespace TEXT,"
+                + "seqId TEXT,"
+                + "seq BIGINT, "
+                + "val BLOB,"
+                + "PRIMARY KEY(namespace, seqId, seq))")
+        boolean createTable();
+
+        @SqlUpdate("INSERT OR REPLACE INTO paxosLog (namespace, seqId, seq, val) VALUES ("
+                + ":namespace.value, :seqId, :seq, :value)")
+        boolean writeRound(
+                @BindPojo("namespace") Client namespace,
+                @Bind("seqId") String seqId,
+                @Bind("seq") long seq,
+                @Bind("value") byte[] value);
+
+        @SqlQuery("SELECT val FROM paxosLog WHERE namespace = :namespace.value AND seqId = :seqId AND seq = :seq")
         @SingleValue
-        byte[] readRound(@Define("table") String table, @Bind("seq") long seq);
+        byte[] readRound(@BindPojo("namespace") Client namespace, @Bind("seqId") String seqId, @Bind("seq") long seq);
 
-        @SqlQuery("SELECT MIN(seq) FROM <table>")
-        OptionalLong getLeastLogEntry(@Define("table") String table);
+        @SqlQuery("SELECT MIN(seq) FROM paxosLog WHERE namespace = :namespace.value AND seqId = :seqId")
+        OptionalLong getLeastLogEntry(@BindPojo("namespace") Client namespace, @Bind("seqId") String seqId);
 
-        @SqlQuery("SELECT MAX(seq) FROM <table>")
-        OptionalLong getGreatestLogEntry(@Define("table") String table);
+        @SqlQuery("SELECT MAX(seq) FROM paxosLog WHERE namespace = :namespace.value AND seqId = :seqId")
+        OptionalLong getGreatestLogEntry(@BindPojo("namespace") Client namespace, @Bind("seqId") String seqId);
 
-        @SqlUpdate("DELETE FROM <table> WHERE seq <= :seq")
-        boolean truncate(@Define("table") String table, @Bind("seq") long seq);
+        @SqlUpdate("DELETE FROM paxosLog WHERE namespace = :namespace.value AND seqId = :seqId AND seq <= :seq")
+        boolean truncate(@BindPojo("namespace") Client namespace, @Bind("seqId") String seqId, @Bind("seq") long seq);
 
-        @SqlBatch("INSERT OR REPLACE INTO <table> (seq, val) VALUES (:round.sequence, :round.valueBytes)")
-        <V extends Persistable & Versionable> boolean[] writeBatchOfRounds(@Define("table") String table,
+        @SqlBatch("INSERT OR REPLACE INTO paxosLog (namespace, seqId, seq, val) VALUES ("
+                + ":namespace.value, :seqId, :round.sequence, :round.valueBytes)")
+        <V extends Persistable & Versionable> boolean[] writeBatchOfRounds(
+                @BindPojo("namespace") Client namespace,
+                @Bind("seqId") String seqId,
                 @BindPojo("round") Iterable<PaxosRound<V>> rounds);
+
+        @SqlQuery("SELECT DISTINCT(namespace) FROM paxosLog")
+        List<String> getAllNamespaces();
```

Comment:
As in the other PR, should be 'Set'

Reply:
Also would be nice if this returned 'Set<Client>' but doesn't seem to be trivial to implement, so this is fine

Reply:
Changing to Set works. As discussed a full change to `Set<Client>` is nontrivial